Search mock data

michael-brennan2005 opened the pull request:
Adds club description + club tag generation via ChatGPT. 

Some notes:
* I rewrote the club scraper to a single python script (sorry @garrettladley). Club scraping is a one-off, non resource intensive task with a lot of string and API fiddling; ultimately I think it makes much more sense for this to be a python script than a Rust project, especially considering that this script is less LoC (470 lines of rust -> 226 lines of python), and that it was faster to write this script from scratch than it was to add the feature to the existing Rust project (Rust ownership semantic hell).
* ChatGPT will generate invalid JSON for club description prompts very rarely (only 2 out of ~40 on one run) and invalid JSON for club tags a lot (half of the tags it generates are ones that weren't given as tags to use).  If this is something we want to reduce w/ prompt engineering or something we can, but its very rare for club descriptions and only 50% of tags being valid is still a lot of tags (~1400 for ~600 clubs on one run), so I think its OK for what this data is intended for. 
* Using ChatGPT to generate this data is very slow - since prompts can't be batched and thus only one description or one set of tags can be made per request - and very expensive - during making this script I used $0.66 of credits, which is 11x more than the credits used for all of the embeddings we've made from the start of this project to now. This is why I've commited the mock.sql, so people can just use that as provided, and we only regenerate the mock.sql if the script changes. Trying to batch the descriptions and/or tag creation through prompt engineering is possible, but, given that this is a one-off script and that we're running into invalid JSON for very simple requests (the description and set of tags generation), imo that isn't something that's worth spending time on.

michael-brennan2005 committed:
Mock data script + mock.sql

michael-brennan2005 committed:
readme + requirements + fixes

michael-brennan2005 committed:
Remove old mock data CI

garrettladley commented on .github/workflows/mock_data.yml:
we should still have CI for this, I'll open an issue and work on that

garrettladley commented on frontend/sac-mobile/expo-env.d.ts:
relic from old frontend directory organization. can we remove?

garrettladley commented on mock_data/main.py:
```diff
@@ -0,0 +1,226 @@
+        TAGS_STRING += f"\"{tag}\", "
+TAGS_STRING = TAGS_STRING[:-2] # remove the last ", "
+
+def description_prompt(name, preview):
```
i'm a big type hint head, please forgive these nits

```suggestion
def description_prompt(name: str, preview: str) -> str:
```

garrettladley commented on mock_data/main.py:
```diff
@@ -0,0 +1,226 @@
+    Your output should just be the JSON with the generated description, no other text. Thank you!
+    """
+
+def tags_prompt(name, description):
```
```suggestion
def tags_prompt(name: str, description: str) -> str:
```

garrettladley commented on mock_data/main.py:
```diff
@@ -0,0 +1,226 @@
+print(f"Bad club descriptions (JSON not created properly): {bad_club_descriptions}")
+
+# Generate the UUIDS for Categories, Tags, and a table for tags to categories.
+category_uuids = {}
```
```suggestion
category_uuids = {category: str(uuid4()) for category in MOCK_CATEGORIES_AND_TAGS}
```

garrettladley commented on mock_data/main.py:
```diff
@@ -0,0 +1,226 @@
+
+# Generate the UUIDS for Categories, Tags, and a table for tags to categories.
+category_uuids = {}
+for category in MOCK_CATEGORIES_AND_TAGS:
```
```suggestion
```

garrettladley commented on mock_data/main.py:
```diff
@@ -0,0 +1,226 @@
+# Generate the UUIDS for Categories, Tags, and a table for tags to categories.
+category_uuids = {}
+for category in MOCK_CATEGORIES_AND_TAGS:
+    category_uuids[category] = str(uuid4())
```
```suggestion
```

garrettladley commented on mock_data/main.py:
```diff
@@ -0,0 +1,226 @@
+for category in MOCK_CATEGORIES_AND_TAGS:
+    category_uuids[category] = str(uuid4())
+
+tag_uuids = {}
```
```suggestion
tag_uuids = {tag: str(uuid4()) for tag in MOCK_CATEGORIES_AND_TAGS.values() for tag in tags}
```

garrettladley commented on mock_data/main.py:
```diff
@@ -0,0 +1,226 @@
+    category_uuids[category] = str(uuid4())
+
+tag_uuids = {}
+tags_to_categories = {}
```
```suggestion
tags_to_categories = {tag: category for category, tags in MOCK_CATEGORIES_AND_TAGS.items() for tag in tags}
```

garrettladley commented on mock_data/main.py:
```diff
@@ -0,0 +1,226 @@
+
+tag_uuids = {}
+tags_to_categories = {}
+for category, tags in MOCK_CATEGORIES_AND_TAGS.items():
```
```suggestion
```

garrettladley commented on mock_data/main.py:
```diff
@@ -0,0 +1,226 @@
+tag_uuids = {}
+tags_to_categories = {}
+for category, tags in MOCK_CATEGORIES_AND_TAGS.items():
+    for tag in tags:
```
```suggestion
```

garrettladley commented on mock_data/main.py:
```diff
@@ -0,0 +1,226 @@
+tags_to_categories = {}
+for category, tags in MOCK_CATEGORIES_AND_TAGS.items():
+    for tag in tags:
+        tag_uuids[tag] = str(uuid4())
```
```suggestion
```

garrettladley commented on mock_data/main.py:
```diff
@@ -0,0 +1,226 @@
+for category, tags in MOCK_CATEGORIES_AND_TAGS.items():
+    for tag in tags:
+        tag_uuids[tag] = str(uuid4())
+        tags_to_categories[tag] = category 
```
```suggestion
```

garrettladley reviewed (commented):
nitty nitty pythonista changes, great work @michael-brennan2005 ! 🌞

garrettladley committed:
Merge branch 'main' into search-mock-data

garrettladley commented on mock_data/main.py:
```diff
@@ -0,0 +1,226 @@
+    ]
+}
+
+TAGS_STRING = "" # for tags_prompt
```
```suggestion
TAGS_STRING = ", ".join(f"\"{tag}\"" for tag in sum(MOCK_CATEGORIES_AND_TAGS.values(), []))
```

garrettladley commented on mock_data/main.py:
```diff
@@ -0,0 +1,226 @@
+}
+
+TAGS_STRING = "" # for tags_prompt
+for key, value in MOCK_CATEGORIES_AND_TAGS.items():
```
```suggestion
```

garrettladley commented on mock_data/main.py:
```diff
@@ -0,0 +1,226 @@
+
+TAGS_STRING = "" # for tags_prompt
+for key, value in MOCK_CATEGORIES_AND_TAGS.items():
+    for tag in value:
```
```suggestion
```

garrettladley commented on mock_data/main.py:
```diff
@@ -0,0 +1,226 @@
+TAGS_STRING = "" # for tags_prompt
+for key, value in MOCK_CATEGORIES_AND_TAGS.items():
+    for tag in value:
+        TAGS_STRING += f"\"{tag}\", "
```
```suggestion
```

garrettladley commented on mock_data/main.py:
```diff
@@ -0,0 +1,226 @@
+for key, value in MOCK_CATEGORIES_AND_TAGS.items():
+    for tag in value:
+        TAGS_STRING += f"\"{tag}\", "
+TAGS_STRING = TAGS_STRING[:-2] # remove the last ", "
```
```suggestion
```

michael-brennan2005 committed:
code review changes

michael-brennan2005 committed:
uhhh weird git issues

michael-brennan2005 committed:
Merge branch 'search-mock-data' of github.com:GenerateNU/sac into search-mock-data

michael-brennan2005 committed:
Mock.sql change

Acc. deleted the old one, add it back and also change the name to be
something longer so ppl are less likely to overwrite it

michael-brennan2005 commented:
@garrettladley Done

garrettladley committed:
untrack frontend/sac-mobile/expo-env.d.ts
